extensions/Worker.Extensions.CosmosDB/src/Config/CosmosDBBindingOptions.cs

```diff
@@ -32,31 +40,33 @@ internal virtual CosmosClient GetClient(string preferredLocations = "")
                 throw new ArgumentNullException(nameof(ConnectionName));
             }
 
-            string cacheKey = BuildCacheKey(ConnectionName!, preferredLocations);
-
-            CosmosClientOptions cosmosClientOptions = new ()
+            if (CosmosExtensionOptions is null)
             {
-                ConnectionMode = ConnectionMode.Gateway
-            };
+                CosmosExtensionOptions = new CosmosDBExtensionOptions();
+            }
+
+            string cacheKey = BuildCacheKey(ConnectionName!, preferredLocations);
 
-            if (!string.IsNullOrEmpty(preferredLocations))
+            // Do not override if preferred locations is configured via CosmosClientOptions
+            if (!string.IsNullOrEmpty(preferredLocations) && CosmosExtensionOptions.ClientOptions.ApplicationPreferredRegions is null)
             {
-                cosmosClientOptions.ApplicationPreferredRegions = Utilities.ParsePreferredLocations(preferredLocations);
+                CosmosExtensionOptions.ClientOptions.ApplicationPreferredRegions = Utilities.ParsePreferredLocations(preferredLocations);
             }
 
-            if (Serializer is not null)
+            // Do not override if the serializer is configured via CosmosClientOptions
+            if (Serializer is not null && CosmosExtensionOptions.ClientOptions.Serializer is null)
             {
-                cosmosClientOptions.Serializer = Serializer;
+                CosmosExtensionOptions.ClientOptions.Serializer = Serializer;
```

Comment:
Can we configured this in the options pipeline instead?

Reply:
Done!